bin.js

```diff
@@ -5,8 +5,13 @@ var os = require('os')
 var path = require('path')
 
 if (!buildFromSource()) {
-  proc.exec('node-gyp-build-test', function (err) {
-    if (err) preinstall()
+  proc.exec('node-gyp-build-test', function (err, stdout, stderr) {
+    if (err) {
+      if (!stderr.includes('No native build was found for')) {
+        console.error('node-gyp-build require test failed:', stderr)
```

Comment:
Isn't `stderr` a buffer here? Meaning this would print e.g. `Buffer <..>`

Reply:
Unsure about the wording here also. The reason it’s silent atm is that this is a fallback step, ie prebuild failed, let’s compile. Introducing an “error” message might confuse users similar to how optional deps confuse.

What about a verbose flag or something similar to build-from-source that spits all of this out?

Reply:
@vweevers By default it's a string but _can_ be a buffer

> The stdout and stderr arguments passed to the callback will contain the stdout and stderr output of the child process. By default, Node.js will decode the output as UTF-8 and pass strings to the callback. The encoding option can be used to specify the character encoding used to decode the stdout and stderr output. If encoding is 'buffer', or an unrecognized character encoding, Buffer objects will be passed to the callback instead.

@mafintosh A verbose flag would do the trick. What I'm after is to somehow get more information on what failed (yeah the wording isn't perfect by any means).